src/_P016_IR.ino

```diff
@@ -137,16 +137,26 @@ boolean Plugin_016(byte function, struct EventStruct *event, String& string)
           irReceiver->setUnknownThreshold(kMinUnknownSize); // Ignore messages with less than minimum on or off pulses.
           irReceiver->enableIRIn(); // Start the receiver
         }
-        if (irReceiver != 0 && irPin == -1)
+        if (irReceiver != 0 && irPin == -1) // what is the purpose of this?
         {
-          Serial.println(F("IR Removed"));
           irReceiver->disableIRIn();
           delete irReceiver;
           irReceiver = 0;
         }
         success = true;
         break;
       }
+          case PLUGIN_EXIT:
+      {
+        {
+          irReceiver->disableIRIn();
```

Comment:
You should check first if `irReceiver != 0`
If it is a null-pointer, de-referencing it will cause a crash (using `->` on a pointer object)